lib/chef/client.rb

```diff
@@ -515,6 +517,48 @@ def setup_run_context
     end
 
     #
+    # Adds a required recipe as specified by the Chef Server
+    #
+    # @return The modified run context
+    #
+    # @api private
+    #
+    # TODO: @rest doesn't appear to be used anywhere outside
+    # of client.register except for here. If it's common practice
+    # to create your own rest client, perhaps we should do that
+    # here but it seems more appropriate to reuse one that we
+    # know is already created. for ease of testing, we'll pass
+    # the existing rest client in as a parameter
```

Comment:
you should probably use `rest_clean` here, so that JSON request/response is sanitized

we probably should have converted everything over to `rest_clean` for Chef-13